Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -601,7 +601,14 @@ public struct RenderNodeTranslator: SemanticVisitor {
         collectedTopicReferences.append(contentsOf: hierarchyTranslator.collectedTopicReferences)
         node.hierarchy = hierarchy
         
-        node.variants = variants(for: documentationNode)
+        // Emit variants only if we're not compiling an article-only catalog to prevent renderers from
+        // advertising the page as "Swift", which is the language DocC assigns to pages in article only pages.
+        // (github.com/apple/swift-docc/issues/240).
+        if let topLevelModule = context.soleRootModuleReference,
```

Comment:
I'm a _little_ concerned about this scaling well in the future. Here's a few questions that came to mind- let me know what you think.

- Should we consider just not emitting variants for documentation nodes that only contain a single variant? Is there a use case for the variant then?

- Why are we only fixing this issue for TechnologyRoot catalogs? Should we fix this for _any_ article that doesn't define a language?

- In the future when we add support for articles with multi-language variants won't this break multi-language articles that are authored in TechnologyRoot catalogs?

Reply:
These are great questions!

> Should we consider just not emitting variants for documentation nodes that only contain a single variant? Is there a use case for the variant then?

This is still useful for renderers to know what language the page is available in for pages originating from frameworks, so that they can display e.g., "Language: Swift" or "Language: Objective-C" in the UI. From the render JSON perspective, the data is "correct", i.e., the page _does_ have that variant. If we decide to not display the language information for single-variant pages, then I'd argue the change should be made in DocC Render, or at least controlled by a separate property in the JSON.

> Why are we only fixing this issue for TechnologyRoot catalogs? Should we fix this for any article that doesn't define a language?

If associated with a module, articles inherit the module's languages, per #81.

> In the future when we add support for articles with multi-language variants won't this break multi-language articles that are authored in TechnologyRoot catalogs?

If we add support for specifying what languages an article is available in, then yes, this logic will need to be updated and we'll likely want to implement a special `undefined` language per #240. But I think we should consider these changes as part of that later fix. I'm also planning on cherry-picking this onto 5.7, so I'd like to minimize risk, and introducing a new special `undefined` language would be a riskier change (e.g., we'd need to update the navigator index logic).

Reply:
That all seems reasonable to me. Thank you!